### DISCUSSION TITLE
https api changed?

### DISCUSSION THREAD

**Discussion by baswelsink** [cesanta/mongoose#2341] [Ask questions, get answers] [answered]
Hi, I was trying to use https for mongoose following this guide https://mongoose.ws/documentation/tutorials/http-client/#build-with-tls-support. However the implementation seems to have changed quite a bit? 

When I compile and run I get the message TLS is not enabled. 

![Screenshot 2023-08-12 160401](https://github.com/cesanta/mongoose/assets/20791767/6721a4e8-acd7-4838-97c3-0055df593163)

This is how I try to enable TLS, not sure if this is enough... 
![Screenshot 2023-08-12 160519](https://github.com/cesanta/mongoose/assets/20791767/60d5f6ac-dead-4f7c-b085-e83c88cb084e)

Could anybody help with this please? :)

**Comment by scaprile** [comment, marked as answer]
Since you don't say, I will assume that you are trying to build using Github HEAD.
We are transitioning our TLS API, adding new features and libs; at this very moment some things do not work with OpenSSL and our docs and examples in HEAD are out of sync. The message you post occurs when there is no TLS code in Mongoose, the way to enable TLS has also changed and by building this way you are not enabling it.
The docs are correct for 7.11 if you pick it up from the releases section and use the examples there; I suggest you start with 7.11 and later migrate to HEAD.
https://github.com/cesanta/mongoose/releases/tag/7.11
https://mongoose.ws/documentation/tutorials/tls/
Check the examples, the http-client example already does what you are trying to do:
https://github.com/cesanta/mongoose/tree/7.11/examples/http-client
https://mongoose.ws/tutorials/http-client/

If you still want to build using HEAD, check the examples and the source code, the API is still not documented as it is not stable and might change again.
https://github.com/cesanta/mongoose/tree/master/examples/http-client
https://github.com/cesanta/mongoose/blob/8c03aa280130ec28d3d0682acac3d0ebe3266ee6/src/tls.h#L3-L11
`make clean all CFLAGS_EXTRA="-DMG_TLS=MG_TLS_OPENSSL -lssl -lcrypto"`

Note: I assume you  are looping when calling `mg_mgr_poll()`; otherwise it wouldn't work.

**Reply by baswelsink** [reply]
Hi @scaprile, thanks a lot for the quick answer! Yes indeed currently I'm using head. Will use the version you suggested, thanks!

**Reply by baswelsink** [reply]
Btw, indeed I was not using a while loop because I didn't want to continuously keep polling the same URL, so I assume from your answer also if you just want to make a one time get request you need to keep using the while loop?

**Reply by scaprile** [reply]
> Check the examples, the http-client example already does what you are trying to do:
https://github.com/cesanta/mongoose/tree/7.11/examples/http-client
https://mongoose.ws/tutorials/http-client/

https://github.com/cesanta/mongoose/blob/5c8e05a4fc34c5309ceea887a7e60b895b2100d7/examples/http-client/main.c#L59-L72

https://mongoose.ws/documentation/#connections-and-event-manager

**Reply by baswelsink** [reply]
Ah perfect, seems I didn't check properly. Thanks!
